examples/rmt_transceiver.rs

```diff
@@ -0,0 +1,113 @@
+//! W (1398) mcpwm_test: level0 = High   dur0 = PulseTicks(410)   level1 = Low   dur1 = PulseTicks(210)
+//! W (1408) mcpwm_test: level0 = High   dur0 = PulseTicks(210)   level1 = Low   dur1 = PulseTicks(0)
+//! W (1918) mcpwm_test: Rx Loop
+//! W (1918) mcpwm_test: level0 = High   dur0 = PulseTicks(620)   level1 = Low   dur1 = PulseTicks(620)
+//! W (1918) mcpwm_test: level0 = High   dur0 = PulseTicks(620)   level1 = Low   dur1 = PulseTicks(620)
+//! W (1928) mcpwm_test: level0 = High   dur0 = PulseTicks(210)   level1 = Low   dur1 = PulseTicks(410)
+//! W (1938) mcpwm_test: level0 = High   dur0 = PulseTicks(410)   level1 = Low   dur1 = PulseTicks(210)
+//! W (1948) mcpwm_test: level0 = High   dur0 = PulseTicks(210)   level1 = Low   dur1 = PulseTicks(0)
+//! E (2378) mcpwm_test: Tx Loop
+
+use esp_idf_sys::{self as _}; // If using the `binstart` feature of `esp-idf-sys`, always keep this module imported
+
+use log::*;
+
+use esp_idf_hal::peripherals::Peripherals;
+use std::thread;
+use std::time::Duration;
+
+use esp_idf_hal::rmt::{
+    config::ReceiveConfig, config::TransmitConfig, FixedLengthSignal, Peripheral, PinState, Pulse,
+    PulseTicks, Receive, Transmit, CHANNEL0, CHANNEL2,
+};
+
+fn main() -> anyhow::Result<()> {
+    // Bind the log crate to the ESP Logging facilities
+    esp_idf_svc::log::EspLogger::initialize_default();
+
+    info!("Starting APP!");
+
+    let peripherals = Peripherals::take().unwrap();
+
+    /*
+     *********************** SET UP RMT RECEIVER ******************************
+     */
+    let input_pin = peripherals.pins.gpio36.into_input()?;
+    let rx_rmt_channel: CHANNEL2 = unsafe { Peripheral::new().channel2 };
+    let rx_config = ReceiveConfig::new().idle_threshold(700u16);
+    let mut rx = Receive::new(input_pin, rx_rmt_channel, &rx_config, 1000)?;
+    let _rx_start = rx.start().unwrap();
+
+    let _receive_task = thread::spawn(move || loop {
+        warn!("Rx Loop");
+
+        // See sdkconfig.defaults to determine the tick time value ( default is one tick = 10 milliseconds)
+        // Set ticks_to_wait to 0 for non-blocking
+        let ticks_to_wait = 0;
+        let length = rx.get_rmt_items(ticks_to_wait).unwrap();
+
+        if length != 0 {
+            for n in 0..length / 4 {
+                warn!(
+                    "level0 = {:?}   dur0 = {:?}   level1 = {:?}   dur1 = {:?}",
+                    rx.pulse_pair_vec[n as usize].level0,
+                    rx.pulse_pair_vec[n as usize].duration0,
+                    rx.pulse_pair_vec[n as usize].level1,
+                    rx.pulse_pair_vec[n as usize].duration1
+                );
+            }
+        }
+
+        thread::sleep(Duration::from_millis(500));
+    });
+
+    /*
+     *********************** SET UP RMT TRANSMITTER ******************************
+     */
+    let output_pin = peripherals.pins.gpio25.into_output()?;
+    let tx_rmt_channel: CHANNEL0 = unsafe { Peripheral::new().channel0 };
```

Comment:
Would it be possible to reuse `peripherals` here
```suggestion
    let tx_rmt_channel: CHANNEL0 = peripherals.channel0;
```

Reply:
I was able to replace with the following
```
let tx_rmt_channel: CHANNEL0 = peripherals.rmt.channel0;
```

Reply:
Will update once I get feedback on discussion below